IdentityCore/src/network/request_server_telemetry/error_provider/MSIDTokenResponseErrorProvider.m

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDTokenResponseErrorProvider.h"
+#import "MSIDOAuth2Constants.h"
+
+@implementation MSIDTokenResponseErrorProvider
+
+- (NSString *)errorForResponse:(__unused NSHTTPURLResponse *)httpResponse
+                          data:(__unused NSData *)data
+                       context:(__unused id<MSIDRequestContext>)context
+                         error:(__unused NSError **)error
+{
+    if (httpResponse.statusCode == 400)
```

Comment:
would there be a possibility of reusing our existing error handling and save exact error that our client code generates to make sure we report same error to telemetry and to the app?

I'd like to avoid duplicating error handling....

Reply:
We agreed to send server-side errors as well as client-side. We still need this one to parse server-side errors.

Reply:
Jepp, we need the super class one to parse http error codes, but parsing here looks more like our client side error handling. Can you clarify what exactly you mean by client and server errors?

Reply:
Removed MSIDTokenResponseErrorProvider.